great_expectations/datasource/fluent/interfaces.py

```diff
@@ -250,14 +288,17 @@ def add_batch_config(self, name: str) -> BatchConfig:
         assert isinstance(self.datasource, Datasource)
 
         batch_config = BatchConfig(name=name)
-        batch_config.set_data_asset(self)
-        self.__fields_set__.add("batch_configs")
-        if self.datasource.is_persisted():
-            batch_config = self.datasource.add_batch_config(batch_config)
-            batch_config.set_data_asset(self)
-
         batch_config.set_data_asset(self)
         self.batch_configs.append(batch_config)
+        if self.datasource.data_context:
+            try:
+                batch_config = self.datasource.add_batch_config(batch_config)
+            except DataContextError:
```

Comment:
I'm wondering if we should do this for any exception since we don't know what the caller might catch and what might make the app crash? I guess it depends what exceptions are possible before/after writing to the backend.